driver-async/src/test/functional/com/mongodb/async/client/TransactionFailureTest.java

```diff
@@ -33,12 +33,23 @@ public TransactionFailureTest() {
 
     @Before
     public void setUp() {
-        assumeTrue(canRunTests());
         super.setUp();
     }
 
     @Test(expected = MongoClientException.class)
-    public void testTransactionFails() throws InterruptedException {
+    public void testTransactionFailsOn40Cluster() throws InterruptedException {
+        assumeTrue(serverVersionLessThan("4.0"));
+        doTransaction();
+    }
+
+    @Test(expected = MongoClientException.class)
+    public void testTransactionFailsOnShardedCluster() throws InterruptedException {
+        assumeTrue(isSharded());
```

Comment:
I realize now that this is a useless test, since we intentionally don't run the 3.10.x tests against any 4.2 clusters on Evergreen.

I did verify that it passes locally using a sharded 4.2 cluster, fwiw.